### PR TITLE
issue-reporting: use artifact instead of resource

### PR DESCRIPTION
**Pull request by jensh007**
**What this PR does / why we need it**:
Refactor issue reporting code to use artifacts instead of resources as preparation for checkmarx integration

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator

```

### CODE REVIEW

File: github/compliance/report.py [hunk 1]
```diff
@@ -116,14 +116,14 @@ def _template_vars(
     delivery_dashboard_url: str='',
 ):
     component = result_group.component
-    resource_name = result_group.resource_name
-    resources = [res.resource for res in result_group.results]
+    resource_name = result_group.name
```
Comment (zkdev) [inline github/compliance/report.py:119]: I'm confused, `<ScanResultGroup>.resource_name()` did return the `cm.Resource`.
Later in this PR, you replace it with `<ScanResultGroup>.artifact()` which returns `cm.Artifact`.

`<ScanResultGroup>.name` however contains `{component.name}:{resource.name}` which is not equal to `<ScanResultGroup>.resource_name()`.

File: github/compliance/result.py [hunk 2]
```diff
@@ -57,8 +57,8 @@ def component(self) -> cm.Component:
         return self.results[0].component
 
     @property
-    def resource_name(self) -> cm.Resource:
-        return self.results[0].resource
+    def artifact(self) -> cm.Artifact:
```
Comment (zkdev) [inline github/compliance/result.py:60]: This is the mentioned property change.

**Review by zkdev — CHANGES_REQUESTED** [review]

File: github/compliance/report.py [hunk 3]
```diff
@@ -431,7 +430,7 @@ def component_resource_label(issue: github3.issues.Issue) -> str:
     for result_group in result_groups:
         resource_label = github.compliance.issue.resource_digest_label(
             component=result_group.component,
-            resource=result_group.resource_name,
+            artifact=result_group.name,
```
Comment (zkdev) [inline github/compliance/report.py:433]: Same here, I would rather expect the newly introduced `<ScanResultGroup>.artifact()`.

File: github/compliance/report.py (hunk 1 above)
Comment (jensh007) [reply github/compliance/report.py]: fixed in next commit

File: github/compliance/result.py (hunk 2 above)
Comment (jensh007) [reply github/compliance/result.py:60]: fixed in next commit

File: github/compliance/report.py (hunk 3 above)
Comment (jensh007) [reply github/compliance/report.py]: fixed in next commit

File: github/compliance/report.py [hunk 4]
```diff
@@ -95,16 +95,16 @@ def pluralise(prefix: str, count: int):
         | -- | -- |
         | Component | {component.name} |
         | Component-Version | {component.version} |
-        | Resource  | {resources[0].name} |
-        | {pluralise('Resource-Version', len(resources))}  | {resource_versions} |
+        | Resource  | {artifacts[0].name} |
```
Comment (ccwienk) [inline github/compliance/report.py:98]: either rename Resource -> Art(e|i)fact, or choose depending on type

**Review by ccwienk — CHANGES_REQUESTED** [review]

**Pull request merged**